### DISCUSSION TITLE
I am getting error in api when send data in json format on live server in Adonis 5.

### DISCUSSION THREAD

**Discussion by pk4all** [adonisjs/core#1403] [Help] [answered]
Hi Team,
I am using AWS S3 for my api hosting. Everything good on my local but on  live server I am getting error "safeParse is not a function" when I have sent data in Json format. It's working good if I have send data in form-data format. I am using adonis 5 and on server I am using production build files.

My Data format like 
``` javascript
{"email":"hello@hello.com","mobile":"1234567890","verification": true}

```
Get data from request
```javascript
   const data=request.all()
    const email=data.email
    const mobile=data.mobile
```

Please suggest me what am I missing?

**Comment by thetutlage** [comment, marked as answer]
Can you share your `package.json` file?

**Reply by pk4all** [reply]
Here is my package.json
```javascript
{
  "name": "seekex",
  "version": "0.0.0",
  "private": true,
  "scripts": {
    "build": "node ace build",
    "start": "node ace serve --watch"
  },
  "devDependencies": {
    "@adonisjs/assembler": "^2.1.3",
    "adonis-preset-ts": "^1.0.4",
    "pino-pretty": "^4.0.1",
    "typescript": "^3.9.6",
    "youch": "^2.0.10",
    "youch-terminal": "^1.0.0"
  },
  "dependencies": {
    "@adonisjs/ace": "^6.9.3",
    "@adonisjs/auth": "^4.2.4",
    "@adonisjs/core": "^5.0.0-preview-rc-1.9",
    "@adonisjs/fold": "^6.3.5",
    "@adonisjs/lucid": "^8.2.2",
    "@adonisjs/session": "^3.0.4",
    "@adonisjs/view": "^2.0.7",
    "aws-sdk": "^2.716.0",
    "axios": "^0.19.2",
    "filesize": "^6.1.0",
    "mysql": "^2.18.1",
    "proxy-addr": "^2.0.6",
    "razorpay": "^2.0.6",
    "reflect-metadata": "^0.1.13",
    "request": "^2.88.2",
    "source-map-support": "^0.5.19"
  }
}
```

**Reply by thetutlage** [reply]
Run the following command to upgrade packages

```sh
npm i @adonisjs/core@alpha @adonisjs/lucid@alpha
```

**Reply by pk4all** [reply]
Thanks for response.
I have run command now I am getting another errors.
```javascript
 "errors": {
        "code": "MODULE_NOT_FOUND",
        "requireStack": [
            "/var/www/html/seekex/node_modules/@adonisjs/hash/build/src/Drivers/Bcrypt.js",
            "/var/www/html/seekex/node_modules/@adonisjs/hash/build/src/Hash/index.js",
            "/var/www/html/seekex/node_modules/@adonisjs/hash/build/providers/HashProvider.js",
            "/var/www/html/seekex/node_modules/@poppinss/utils/build/src/esmRequire.js",
            "/var/www/html/seekex/node_modules/@poppinss/utils/build/index.js",
            "/var/www/html/seekex/node_modules/@adonisjs/fold/build/src/Ioc/index.js",
            "/var/www/html/seekex/node_modules/@adonisjs/fold/build/index.js",
            "/var/www/html/seekex/node_modules/@adonisjs/core/build/src/Ignitor/Bootstrapper/index.js",
            "/var/www/html/seekex/node_modules/@adonisjs/core/build/src/Ignitor/Ace/AppCommands.js",
            "/var/www/html/seekex/node_modules/@adonisjs/core/build/src/Ignitor/Ace/index.js",
            "/var/www/html/seekex/node_modules/@adonisjs/core/build/src/Ignitor/index.js",
            "/var/www/html/seekex/server.js"
        ]
    },
    "resMsg": "Invalid Username or Password",
```

**Reply by RomainLanz** [reply]
You need to install the Bcrypt Driver.

```
npm i phc-bcrypt
```

**Reply by kennedyrs** [reply]
Maybe your node version, is it the same as your localhost?

**Reply by pk4all** [reply]
@RomainLanz Thanks My problem is solved.

**Comment by kennedyrs** [comment]
Have you modified the bodyparser.ts?

**Reply by pk4all** [reply]
@kennedyrs No not modified. It's working on my local machine, But not working on My AWS server.

**Comment by pikitgb** [comment]
@pk4all thanks a lot for reporting this issue, I went crazy for like 2 hours looking for an explanation why this started to happen, 

I began to experience the same issue today, and probably started to happen about two or three days ago, but i was aware of the issue today after seeing logs with error. I didn't performed any code update, or no environment change. I'm using Heroku and Adonis 5 also and my app was already deployed previously, it's been working fine for about 2 or 3 months without any issues.

So I think this was caused because probably Heroku (in my case) uses internally AWS EC2 instances and RDS instances for the Database given that i'm using PG for the DB and the error seems to be after querying the database. Very strange error indeed.

Anyways, kudos and love to @thetutlage and @RomainLanz using `npm i @adonisjs/core@alpha @adonisjs/lucid@alpha` and `npm i phc-bcrypt` also did the magic for me.
